service/collector/collector.go

```diff
@@ -95,42 +120,46 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 				continue
 			}
 
-			a, ok := event.Object.(*providerv1alpha1.AzureConfig)
+			customObject, ok := event.Object.(*providerv1alpha1.AzureConfig)
```

Comment:
Thinking it over again, we want to list/expose all deployments in the account. Consider a case where a CR is deleted, but the associated resources are not cleaned up correctly.

Reply:
That is not possible because all deployments are within the resource group. When you want to have this safety check we only need to check the resource groups but this is out of scope of my actual work here.

Reply:
I don't follow. Could we list the RGs, and then the deployments?